### PR TITLE
decompile contest_ai.c

### PR DESCRIPTION
**Pull request by ProjectRevoTPP**

### CODE REVIEW

**Comment by ProjectRevoTPP** [comment]
This file is so repetitive This file is so repetitive This file is so repetitive This file is so repetitive This file is so repetitive This file is so repetitive This file is so repetitive

File: src/contest_ai.c [hunk 1]
```diff
@@ -0,0 +1,1770 @@
...
+#include "random.h"
+#include "ewram.h"
+
+extern u8 AreMovesContestCombo(u16, u16);
+extern bool8 sub_80B214C(u8);
+extern bool8 Contest_IsMonsTurnDisabled(u8);
+
+enum
+{
+    ContestAI_SettingUp,
```
Comment (huderlem) [inline src/contest_ai.c]: enums should be ALL CAPS with underscores

File: src/contest_ai.c [hunk 2]
```diff
@@ -0,0 +1,1770 @@
...
+
+static void ContestAI_DoAIProcessing(void);
+static bool8 sub_8128A7C(u8);
+static void sub_812ACA4(u8 *);
+static u8 sub_812ACC8(void);
+
+void ContestAI_ResetAI(u8 var)
+{
+    int i;
+    memset(eContestAI, 0, sizeof(struct ContestAIInfo));
```
Comment (huderlem) [inline src/contest_ai.c]: can this just be `eContestAI = {0};`?

File: src/contest_ai.c [hunk 3]
```diff
@@ -0,0 +1,1770 @@
...
+        gAIScriptPtr = T1_READ_PTR(gAIScriptPtr + 1);
+    else
+        gAIScriptPtr += 5;
+}
+
+static void ContestAICmd_unk_05(void)
+{
+    ContestAICmd_get_turn();
+
+    if((s16)eContestAI->scriptResult != gAIScriptPtr[0])
```
Comment (huderlem) [inline src/contest_ai.c]: Is the `(s16)` cast in all of these command funcs needed to match?

File: src/contest_ai.c (hunk 2 above)
Comment (ProjectRevoTPP) [reply src/contest_ai.c]: I tried. No.

**Pull request merged**